src/Symfony/Component/DependencyInjection/ContainerBuilder.php

```diff
@@ -126,7 +126,7 @@ public function __construct(ParameterBagInterface $parameterBag = null)
         parent::__construct($parameterBag);
 
         $this->trackResources = interface_exists('Symfony\Component\Config\Resource\ResourceInterface');
-        $this->setDefinition('service_container', (new Definition(ContainerInterface::class))->setSynthetic(true));
+        $this->setDefinition('service_container', (new Definition(parent::class))->setSynthetic(true)->setPublic(false));
```

Comment:
nope: the base container class can be changed when the container is dumped
but of course, the base class of ContainerBuilder cannot
so everything is fine, and this change should be reverted